src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeName.cs

```diff
@@ -401,6 +436,25 @@ public static bool TryParse(ReadOnlySpan<char> typeName, [NotNullWhen(true)] out
             return result is not null;
         }
 
+        /// <summary>
+        /// Converts any escaped characters in the input type name or namespace.
+        /// </summary>
+        /// <param name="name">The input string containing the name to convert.</param>
+        /// <returns>A string of characters with any escaped characters converted to their unescaped form.</returns>
+        /// <remarks>
+        /// <para>The unescaped string can be used for looking up the type name or namespace in metadata.</para>
+        /// <para>This method removes escape characters even if they precede a character that does not require escaping.</para>
+        /// </remarks>
+        public static string Unescape(string name)
+        {
+            if (name is null)
+            {
+                TypeNameParserHelpers.ThrowArgumentNullException(nameof(name));
```

Comment:
```suggestion
                Throw.ArgumentNull(nameof(name));
```
Can we use the pre-existing helper from System.Reflection.Metadata?

Reply:
I'm not sure, the `Throw` file is not available in Corelib. I can add an `#if SYSTEM_REFLECTION_METADATA` and just `throw` otherwise, if you prefer that.

Reply:
Or I can make `TypeName.Unescape` exclusive to SRM, and use `TypeNameParserHelpers.Unescape` in Corelib.

Reply:
Ah, I forgot about sharing with CoreLib.